algosdk/future/transaction.py

```diff
@@ -434,47 +446,196 @@ def dictify(self):
+        selkey (str): VRF public key in base64
+        votefst (int): first round to vote
+        votelst (int): last round to vote
+        votekd (int): vote key dilution
+        note (bytes, optional): arbitrary optional bytes
+        lease (byte[32], optional): specifies a lease, and no other transaction
+            with the same sender and lease can be confirmed in this
+            transaction's valid rounds
+        rekey_to (str, optional): additionally rekey the sender to this address
+
+    Attributes:
+        sender (str)
+        fee (int)
+        first_valid_round (int)
+        last_valid_round (int)
+        note (bytes)
+        genesis_id (str)
+        genesis_hash (str)
+        group(bytes)
+        votepk (str)
+        selkey (str)
+        votefst (int)
+        votelst (int)
+        votekd (int)
+        type (str)
+        lease (byte[32])
+        rekey_to (str)
+    """
+
+    def __init__(self, sender, sp, votekey, selkey, votefst,
+                 votelst, votekd, note=None,
+                 lease=None, rekey_to=None):
+        KeyregTxn.__init__(self, sender, sp, votekey, selkey, votefst, votelst,
+                           votekd, note, lease, rekey_to, nonpart=False)
+        self.votepk = votekey
+        self.selkey = selkey
+        self.votefst = votefst
+        self.votelst = votelst
+        self.votekd = votekd
+        if votekey is None:
+            raise error.KeyregOnlineTxnInitError("votekey")
+        if selkey is None:
+            raise error.KeyregOnlineTxnInitError("selkey")
+        if votefst is None:
+            raise error.KeyregOnlineTxnInitError("votefst")
+        if votelst is None:
+            raise error.KeyregOnlineTxnInitError("votelst")
+        if votekd is None:
+            raise error.KeyregOnlineTxnInitError("votekd")
+        if not sp.flat_fee:
+            self.fee = max(self.estimate_size() * self.fee,
+                           constants.min_txn_fee)
+
     @staticmethod
     def _undictify(d):
-        votekey = None
-        selkey = None
-        votefst = None
-        votelst = None
-        votekd = None
-        nonpart = None
-
-        if "votekey" in d:
-            votekey = base64.b64encode(d["votekey"]).decode()
-        if "selkey" in d:
-            selkey = base64.b64encode(d["selkey"]).decode()
-        if "votefst" in d:
-            votefst = d["votefst"]
-        if "votelst" in d:
-            votelst = d["votelst"]
-        if "nonpart" in d:
-            nonpart = d["nonpart"]
-
+        votekey = base64.b64encode(d["votekey"]).decode()
+        selkey = base64.b64encode(d["selkey"]).decode()
+        votefst = d["votefst"]
+        votelst = d["votelst"]
+        votekd = d["votekd"]
         args = {
             "votekey": votekey,
             "selkey": selkey,
             "votefst": votefst,
             "votelst": votelst,
             "votekd": votekd,
-            "nonpart": nonpart
```

Comment:
If you're curious why `nonpart` is removed here, it's because it already exists in the dictionary.